enumerables.rb

```diff
@@ -0,0 +1,171 @@
+module Enumerable
+  # my_each
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i])
+      i += 1
+    end
+  end
+
+  # my_each_with_index
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < length
+      yield([i], i)
+      i += 1
+    end
+  end
+
+  # my_select
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    i = 0
+    while i < length
+      arr << [i] if yield(self[i])
+      i += 1
+    end
+    arr
+  end
+
+  # my all
+
+  def my_all?(aaa = nil)
+    if aaa
+
+      my_each { |elem| return false unless aaa == elem }
+
+    end
+    unless block_given?
+      i = 0
+      while i < length
+        return false unless self[i]
+
+        i += 1
+      end
+      return true
+    end
+    bool = true
+    i = 0
+    while i < length
+      bool = false unless yield(self[i])
+      i += 1
+    end
+    bool
+  end
+
+  # my any
+
+  def my_any?(aaa = nil)
+    if aaa
+
+      my_each { |elem| return true unless aaa == elem }
+
+    end
+    unless block_given?
+      i = 0
+      while i < length
+        return true unless [i]
+
+        i += 1
+      end
+      return false
+    end
+    bool = false
+    i = 0
+    while i < length
+      bool = true if yield(self[i])
+      i += 1
+    end
+    bool
+  end
+
+  # my none?
+
+  def my_none?(aaa = nil, &prc)
+    !my_any(aaa, &prc) if aaa
+    unless block_given?
+      i = 0
+      while i < length
+        return false unless [i]
+
+        i += 1
+      end
+      return true
+    end
+    bool = true
+    i = 0
+    while i < length
+      bool = false if yield(self[i])
+      i += 1
+    end
+    bool
+  end
```

Comment:
* [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
  ```ruby
    array = [1,2,5,4,7]
    p array.my_none? == array.none? # false
  ```

* [ ] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
  ```ruby
    array = [1,2,5,4,7]
    p array.my_none?(String) == array.none?(String) # false
  ```

* [ ] my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
  ```ruby
    array = %w[dog door rod blade]
    p array.my_none?(/c/) == array.none?(/c/) # false
  ```
  - You have typo, my_any? not my_any :mag: 

* [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
  ```ruby
    array = %w[dog door rod blade]
    p array.my_none?(5) == array.none?(5) # false
  ```